packages/core/src/stepFunctions/workflowStudio/workflowStudioEditorProvider.ts

```diff
@@ -0,0 +1,160 @@
+        const localeTag = `<meta name='locale' content='${locale}'>`
+        const theme = vscode.window.activeColorTheme.kind
+        const isDarkMode = theme === vscode.ColorThemeKind.Dark || theme === vscode.ColorThemeKind.HighContrast
+        const darkModeTag = `<meta name='dark-mode' content='${isDarkMode}'>`
+        let html = `${htmlFileSplit[0]} <head> ${baseTag} ${localeTag} ${darkModeTag} ${htmlFileSplit[1]}`
+
+        const nonce = getRandomString()
+        htmlFileSplit = html.split("script-src 'self'")
+
+        html = `${htmlFileSplit[0]} script-src 'self' 'nonce-${nonce}' ${isLocalDev && localhost} ${htmlFileSplit[1]}`
+        htmlFileSplit = html.split('<body>')
+        const script = await fs.readFileAsString(
+            vscode.Uri.joinPath(this.extensionContext.extensionUri, 'resources', 'js', 'vsCodeExtensionInterface.js')
+        )
+
+        return `${htmlFileSplit[0]} <body> <script nonce='${nonce}'>${script}</script> ${htmlFileSplit[1]}`
+    }
+
+    /**
+     * Called when the custom editor is opened.
+     * @param document:  The document to be displayed in the editor.
+     * @param webviewPanel: The webview panel that the editor should be displayed in.
+     * @param _token: A cancellation token that can be used to cancel the editor.
+     * @returns A promise that resolves when the editor is resolved.
+     */
+    public async resolveCustomTextEditor(
+        document: vscode.TextDocument,
+        webviewPanel: vscode.WebviewPanel,
+        _token: vscode.CancellationToken
+    ): Promise<void> {
+        await telemetry.stepfunctions_openWorkflowStudio.run(async () => {
+            if (!this.webviewHtml) {
+                await this.fetchWebviewHtml()
+            }
+
+            if (clientId === '') {
+                clientId = getClientId(globals.globalState)
+            }
+            // Attempt to retrieve existing visualization if it exists.
+            const existingVisualization = this.managedVisualizations.get(document.uri.fsPath)
+
+            if (existingVisualization) {
+                existingVisualization.showPanel()
+            } else {
+                // Existing visualization does not exist, construct new visualization
+                try {
+                    const fileId = getStringHash(`${document.uri.fsPath}${clientId}`)
+                    const newVisualization = new WorkflowStudioEditor(
+                        document,
+                        webviewPanel,
+                        this.extensionContext,
+                        fileId,
+                        this.getWebviewContent
+                    )
+                    this.handleNewVisualization(document.uri.fsPath, newVisualization)
+                } catch (err) {
+                    void vscode.window.showInformationMessage(i18n('AWS.stepFunctions.workflowStudio.errors.rendering'))
+                    this.logger.error(`WorkflowStudio: Unable to open file. ${err}`)
```

Comment:
If `resolveCustomTextEditor` is called as the result of a vscode command in any form (which is usually the case,) then Toolkit already has a default handler that logs the error and shows a message. It's not necessary and discouraged to repeat that logic in this way, because it leads to an incosistent UX. 

Please review the guidlines here: https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#develop